traffic_ops/traffic_ops_golang/api/generic_crud.go

```diff
@@ -119,19 +119,39 @@ func GenericCreateNameBasedID(val GenericCreator) (error, error, int) {
 	}
 	defer resultRows.Close()
 
+	var name string
 	lastUpdated := tc.TimeNoMod{}
 	rowsAffected := 0
-	for resultRows.Next() {
-		rowsAffected++
-		if err := resultRows.Scan(&lastUpdated); err != nil {
-			return nil, errors.New(val.GetType() + " create scanning: " + err.Error()), http.StatusInternalServerError
+
+	// Only when the type is of serviceCategory, &name is scanned and returned from the DB.
+	// Else return only &lastUpdated.
+	if val.GetType() == "serviceCategory" {
+		for resultRows.Next() {
```

Comment:
You can place the if get type block after for loop, so as to not to repeat the same code twice.

Reply:
Suggestion Implemented